### PR TITLE
Add modals for advanced mode

### LINKED ISSUE #362: Friction for advanced mode
- Make sending using recipient's standard public key in advanced mode harder

Ideas:
1. When user checks the checkbox, have a pop-up warning to make sure users understand.
2. Another pop-up when user clicks Send.
3. Any other ideas?

### PR DESCRIPTION
**Pull request by alexkeating**
Closes #362 

Description:

I added two modals one for when "send using recipient's public key is selected" and another on when the send is clicked. I based the copy on the issue and open to improving it.

[Screencast from 10-11-2022 07:59:35 PM.webm](https://user-images.githubusercontent.com/12705146/195219578-a6d1ad3c-3052-4d51-8830-851525f61875.webm)

### CODE REVIEW

**Review by apbendi — COMMENTED** [review]
Looks and works well @alexkeating, but let's adjust the copy a bit. I'd also like to expand the functionality a bit (sorry for the requirements-on-the-fly here).

* Message title: Remove the exclamation point after "Warning"
* Message Copy: "The receiver will have to enter their account's private key into this app to withdrawal the funds. **Don't use this feature unless you know what you're doing**. [Learn more](http://localhost:8080/faq#how-do-i-send-funds-to-a-user-by-their-address-or-public-key)"
* Functionality: In addition to the warning, add a check box to the second modal displayed after the user clicks "Send" that reads "☐ I acknowledge the risks". The final "Send" button on the modal should be inactive until the user checks the box.

File: frontend/src/i18n/locales/zh-cn.json [hunk 1]
```diff
@@ -78,7 +78,8 @@
     "amount-exceeds-balance": "钱包金额不足",
     "please-complete-form": "请填写表格",
     "wallet-not-connected": "钱包未连接",
-    "umbra-fee-exceeds-balance": "需要的 Umbra 费用超过钱包金额"
+    "umbra-fee-exceeds-balance": "需要的 Umbra 费用超过钱包金额",
+    "advanced-mode-warning": "使用高级模式可能会导致使资金无法恢复的操作"
```
Comment (mds1) [inline frontend/src/i18n/locales/zh-cn.json:82]: cc @garyghayrat for a translation check

File: frontend/src/i18n/locales/zh-cn.json [hunk 2]
```diff
@@ -310,6 +311,9 @@
       "same-addr-as-stealth": "与隐形地址相同",
       "address-it-resolves-to": "它解析为的地址",
       "has-poap-tokens": "具有 POAP 令牌"
+    },
+    "Dialog": {
+      "warning": "警告！"
```
Comment (mds1) [inline frontend/src/i18n/locales/zh-cn.json:316]: @garyghayrat here too

File: frontend/src/pages/AccountSend.vue [hunk 3]
```diff
@@ -144,8 +195,18 @@
       </div>
 
       <!-- Send button -->
+      <!-- Pop modal and send in that -->
```
Comment (mds1) [inline frontend/src/pages/AccountSend.vue:198]: A bit confused by these two comment lines—Is one outdated? Also not sure what "Pop modal and send in that" means 😅

Comment (alexkeating) [reply frontend/src/pages/AccountSend.vue:198]: Lol, oops must have accidentally left a note behind

**Comment by alexkeating** [comment]
> Looks and works well @alexkeating, but let's adjust the copy a bit. I'd also like to expand the functionality a bit (sorry for the requirements-on-the-fly here).
> 
>     * Message title: Remove the exclamation point after "Warning"
> 
>     * Message Copy: "The receiver will have to enter their account's private key into this app to withdrawal the funds. **Don't use this feature unless you know what you're doing**. [Learn more](http://localhost:8080/faq#how-do-i-send-funds-to-a-user-by-their-address-or-public-key)"
> 
>     * Functionality: In addition to the warning, add a check box to the second modal displayed after the user clicks "Send" that reads "☐ I acknowledge the risks". The final "Send" button on the modal should be inactive until the user checks the box.

I updated to reflect these requests

**Review by apbendi — APPROVED** [review]
Updates look great @alexkeating! Thanks.

File: frontend/src/i18n/locales/zh-cn.json [hunk 4]
```diff
@@ -78,7 +78,10 @@
     "amount-exceeds-balance": "钱包金额不足",
     "please-complete-form": "请填写表格",
     "wallet-not-connected": "钱包未连接",
-    "umbra-fee-exceeds-balance": "需要的 Umbra 费用超过钱包金额"
+    "umbra-fee-exceeds-balance": "需要的 Umbra 费用超过钱包金额",
+    "advanced-send-warning": "接收者必须将其帐户的私钥输入此应用程序才能提取资金。除非您知道自己在做什么，否则不要使用此功能",
```
Comment (garyghayrat) [inline frontend/src/i18n/locales/zh-cn.json:82]:
```suggestion
    "advanced-send-warning": "接收者必须将其帐户的私钥输入此应用程序才能提取资金。除非您知道自己在做什么，不要使用此功能",
```

File: frontend/src/i18n/locales/zh-cn.json (hunk 2 above)
Comment (garyghayrat) [reply frontend/src/i18n/locales/zh-cn.json:318]: looks good

File: frontend/src/i18n/locales/en-us.json [hunk 5]
```diff
@@ -78,7 +78,10 @@
     "amount-exceeds-balance": "Amount exceeds wallet balance",
     "please-complete-form": "Please complete the form",
     "wallet-not-connected": "Wallet not connected",
-    "umbra-fee-exceeds-balance": "required for Umbra fee exceeds wallet balance"
+    "umbra-fee-exceeds-balance": "required for Umbra fee exceeds wallet balance",
+    "advanced-send-warning": "The receiver will have to enter their account's private key into this app to withdrawal the funds. Don't use this feature unless you know what you're doing.",
+    "learn-more": "Learn more",
+    "acknowledge-risks": "I acknowledge the risks"
```
Comment (mds1) [inline frontend/src/i18n/locales/en-us.json:84]: _(arbitrary comment location to start a thread)_

The token  warning PR and send button modal have a checkbox to acknowledge, but the pubkey modal just has a button with no checkbox. Personally I like the checkbox so we should add it to be consistent

File: frontend/src/pages/AccountSend.vue [hunk 6]
```diff
@@ -400,6 +486,7 @@ function useSendForm() {
       resetForm();
     } finally {
       isSending.value = false;
+      showAdvancedSendWarning = false;
```
Comment (mds1) [inline frontend/src/pages/AccountSend.vue:489]: Also you define `let showAdvancedSendWarning = ref(false);` but it should be `const showAdvancedSendWarning = ref(false);`
```suggestion
      showAdvancedSendWarning.value = false;
```

File: frontend/src/i18n/locales/en-us.json (hunk 5 above)
Comment (alexkeating) [reply frontend/src/i18n/locales/en-us.json:84]: Adding the checkbox is fine with me.

**Review by mds1 — APPROVED** [review]
LGTM, merge is just waiting for translation confirmation by @garyghayrat (https://github.com/ScopeLift/umbra-protocol/pull/403#discussion_r997172347) and any new QA testing of the changes from @apbendi

**Review by garyghayrat — APPROVED** [review]

File: frontend/src/i18n/locales/en-us.json [hunk 7]
```diff
@@ -78,7 +78,10 @@
     "amount-exceeds-balance": "Amount exceeds wallet balance",
     "please-complete-form": "Please complete the form",
     "wallet-not-connected": "Wallet not connected",
-    "umbra-fee-exceeds-balance": "required for Umbra fee exceeds wallet balance"
+    "umbra-fee-exceeds-balance": "required for Umbra fee exceeds wallet balance",
+    "advanced-send-warning": "The receiver will have to enter their account's private key into this app to withdrawal the funds. Don't use this feature unless you know what you're doing.",
```
Comment (garyghayrat) [inline frontend/src/i18n/locales/en-us.json:82]:
```suggestion
    "advanced-send-warning": "The receiver will have to enter their account's private key into this app to withdraw the funds. Don't use this feature unless you know what you're doing.",
```

**Comment by garyghayrat** [comment]
Having these weird `Bad response` and sometimes `invalid address` errors in the address bar only when `Send using recipient's standard public key' is selected and an amount that exceeds wallet balance or letters are entered.
image

**Comment by alexkeating** [comment]
> Having these weird `Bad response` and sometimes `invalid address` errors in the address bar only when `Send using recipient's standard public key` is selected and an amount that exceeds wallet balance or letters are entered. image

Hmmm, I am not seeing these locally. Is there anything I am doing wrong to reproduce?

[Screencast from 10-20-2022 04:54:49 PM.webm](https://user-images.githubusercontent.com/12705146/197056413-54f3f926-5c34-4b87-a25b-7b28f98a741f.webm)

**Review by apbendi — APPROVED** [review]
Looking and working great @alexkeating. Nice work, thanks for the adjustments!

**Pull request merged**